airflow/providers/yandex/hooks/http_client.py

```diff
@@ -0,0 +1,321 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This file is a copy of https://github.com/ydb-platform/ydb/tree/284b7efb67edcdade0b12c849b7fad40739ad62b/ydb/core/fq/libs/http_api_client
+# It is highly recommended to modify original file first in YDB project and merge it here afterwards
```

Comment:
Vendored parts should be stored into the `_vendor`, may I know the reason why we need vendor this code?

Reply:
> But improving YQ backend is not easy^ we would spent several months to make it.

I don't think this is reason enough.
This feels a bit abusive to vendor this code into Airflow.
We normally vendor in only in rare care cases to prevent backward incompatibilities issues and we do that for a very limited time.

Also, this is not really a simple vendoring. The code that you are asking to vendor is not "static" you may change it thus you will also need to change it in Airflow and release managers will need to accomodate it every time you wish to make a change. 

Given the situation I think you should start a mailing list discussion and explain your request. If needed PMCs can vote on your request and accept/decline it.

Reply:
Thank you for feedback!
Let me add more details. YandexQuery has public HTTP API which is described here: https://cloud.yandex.ru/en/docs/query/api/.
That means that any person can write client to YQ using this protocol in any language he wish and put it in any github repo. So did I. The minor problem is that I had to copy-paste logic since I need those clients in two repos. It is up to YQ backend to preserve backward compatibility. I did not create pypi package for client library since I'm not sure that client API would be stable, but HTTP API will be stable and it is YQ backend responsibility to make it correctly. Moreover new features will appear in protocol, but without breaking compatibility with old client (hopefully). I'm aware of YandexQuery backend plans since I'm member of this team ATM.
Currently I see no problem with this approach. Could you please add more details where you see abusing or some difficulties with this approach?
Thank you in advance for discussion this issue, I believe this is very important from legal perspective as well as from stability and project maintenance simplicity and I would be happy to give more details to you.

Reply:
If this is just the one file, I am not too concerned for licence and copying. These are merely wrappers around a few calls. 

However,  that calls even more for a set of tests there. It would be enough to have a set of test cases where the client is mocked (as we do in many other hooks. This has the nice feature as well that it protects against accidental regressions. if this code is copied from elsewhere, having independent set of hook tests that a) mock client calls b) simulate what the operator does - is ratther benefitial to prevent problems, imagine for example the code that got copied will have a method renamed but you forget to update the operator. If you have a set of Hook unit tests that call all the methods - it wil be cautght there (of course MyPy and linters might catch it as well but this is not always possible).

Reply:
added test with mock http client